### PR TITLE
Make control UIControl subclass instead of UIView (close #18)

### LINKED ISSUE #18: Make RainyRefreshControl subclass of UIControl or UIRefreshControl
Check https://github.com/delannoyk/GIFRefreshControl as an example.

This way we'll be able to use it in the (almost) same way we use `UIRefreshControl`.

### PR DESCRIPTION
**Pull request by dolzhenko-tony**

### CODE REVIEW

File: RainyRefresh/ONXRefreshControl.swift [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        if let superview = superview as? UIScrollView {
+            superview.addObserver(self, forKeyPath: "contentOffset", options: [.new, .old], context: nil)
+        }
+        super.didMoveToSuperview()
+    }
+
+    // MARK: Refresh methods
+
+    public func beginRefreshing() {
+        if let superview = superview as? UIScrollView, !refreshing {
```
Comment (derpoliuk) [inline RainyRefresh/ONXRefreshControl.swift]: It's better to use `guard` here.

File: RainyRefresh/ONXRefreshControl.swift [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            //Aaaaand scrolling
+            superview.setContentOffset(currentOffset, animated: false)
+            superview.setContentOffset(CGPoint(x: 0, y: -inset.top), animated: true)
+            forbidsOffsetChanges = true
+            didBeginRefresh()
+        }
+    }
+    
+    public func endRefreshing() {
+        if let superview = superview as? UIScrollView, refreshing {
```
Comment (derpoliuk) [inline RainyRefresh/ONXRefreshControl.swift]: It's better to use `guard` here.

File: RainyRefresh/ONXRefreshControl.swift [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                    self.didEndRefresh()
+                }
+            }
+        }
+    }
+    
+    // MARK: KVO
+    
+    public override func observeValue(forKeyPath keyPath: String?, of object: Any?, change: [NSKeyValueChangeKey : Any]?, context: UnsafeMutableRawPointer?) {
+        if let superview = superview as? UIScrollView, !changingInset {
```
Comment (derpoliuk) [inline RainyRefresh/ONXRefreshControl.swift]: It's better to use `guard` here.

File: RainyRefresh/ONXRefreshControl.swift [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        commonInit()
+    }
+    
+    private func commonInit() {
```
Comment (derpoliuk) [inline RainyRefresh/ONXRefreshControl.swift]: Let's get rid of `commonInit` and `setup` methods and leave this logic to subclasses. Reason - it is just adds boilerplate code to this class without any functionality.

File: RainyRefresh/RainyRefreshControl.swift [hunk 5]
```diff
@@ -9,148 +9,71 @@
...
+        backgroundView.backgroundColor = bgColor
+        addSubview(backgroundView)
+        
+        backgroundView.showsFPS = false
+        backgroundView.showsNodeCount = false
+        
+        /* Sprite Kit applies additional optimizations to improve rendering performance */
+        backgroundView.ignoresSiblingOrder = true
+        
+        //        if let scene = GameScene.unarchiveFromFile("GameScene") as? GameScene {
```
Comment (derpoliuk) [inline RainyRefresh/RainyRefreshControl.swift]: Do we need this?

File: RainyRefresh/ONXRefreshControl.swift [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+import UIKit
+
+public class ONXRefreshControl: UIControl {
+    
+    private var forbidsOffsetChanges = false
+    private var forbidsInsetChanges = false
+    private var changingInset = false
+    private var refreshing = false
+    private var contentInset: UIEdgeInsets?
+    public var animationDuration = TimeInterval(0.33)
```
Comment (derpoliuk) [inline RainyRefresh/ONXRefreshControl.swift:15]: Do we need make those properties `public`?

**Review by derpoliuk — CHANGES_REQUESTED** [review]

File: RainyRefresh/RainyRefreshControl.swift (hunk 5 above)
Comment (dolzhenko-tony) [reply RainyRefresh/RainyRefreshControl.swift]: removed

File: RainyRefresh/ONXRefreshControl.swift (hunk 3 above)
Comment (dolzhenko-tony) [reply RainyRefresh/ONXRefreshControl.swift]: changed

File: RainyRefresh/ONXRefreshControl.swift (hunk 2 above)
Comment (dolzhenko-tony) [reply RainyRefresh/ONXRefreshControl.swift]: changed

File: RainyRefresh/ONXRefreshControl.swift (hunk 1 above)
Comment (dolzhenko-tony) [reply RainyRefresh/ONXRefreshControl.swift]: changed

File: RainyRefresh/ONXRefreshControl.swift (hunk 4 above)
Comment (dolzhenko-tony) [reply RainyRefresh/ONXRefreshControl.swift]: what about just to leave ```setup``` and place it instead of ```commonInit``` ?

**Review by derpoliuk — APPROVED** [review]

**Comment by derpoliuk** [comment]
Great, thank you! Merging.

**Pull request merged**